src/Datagrid/Pager.php

```diff
@@ -587,6 +587,7 @@ protected function setLastPage($page)
     {
         $this->lastPage = $page;
 
+        // NEXT_MAJOR: Remove this code.
```

Comment:
When trying to access to page 3 when there is only two pages, this code leads to weird behaviour

```
$offset = ($this->getPage() - 1) * $this->getMaxPerPage();
$this->setLastPage((int) ceil($this->getNbResults() / $this->getMaxPerPage()));
```
and
```
$this->setLastPage((int) ceil($this->getNbResults() / $this->getMaxPerPage()));
$offset = ($this->getPage() - 1) * $this->getMaxPerPage();
```
doesn't give the same offset value.

I think we shouldn't set the page value silently.

Reply:
Shouldn't we trigger a deprecation?

Reply:
If I add a deprecation, the developer won't be able to avoid it.
It's not mandatory to call setPage, before setLastPage.